File: Editorial/129_Sum_Root_to_Leaf_Numbers/CSharp/2_Recursive_Preorder_Traversal.cs
public class Solution {
    int rootToLeaf = 0;

    public void Preorder(TreeNode r, int currNumber) {
        if (r != null) {
            currNumber = currNumber * 10 + r.val;
            // if it's a leaf, update root-to-leaf sum
            if (r.left == null && r.right == null)
                rootToLeaf += currNumber;
            Preorder(r.left, currNumber);
            Preorder(r.right, currNumber);
        }
    }

    public int SumNumbers(TreeNode root) {
        Preorder(root, 0);
        return rootToLeaf;
    }
}

File: Editorial/129_Sum_Root_to_Leaf_Numbers/JavaScript/2_Recursive_Preorder_Traversal.js
var sumNumbers = function (root) {
    function preorder(r, currNumber) {
        if (r) {
            currNumber = currNumber * 10 + r.val;
            // if it's a leaf, update root-to-leaf sum
            if (!r.left && !r.right) rootToLeaf += currNumber;
            preorder(r.left, currNumber);
            preorder(r.right, currNumber);
        }
    }
    var rootToLeaf = 0;
    preorder(root, 0);
    return rootToLeaf;
};
